src/engine/Eval_generic.ml

```diff
@@ -246,18 +284,30 @@ and eval_op op values code =
   | G.Gt, [ Float i1; Float i2 ] -> Bool (i1 > i2)
   | G.Gt, [ Int i1; Float i2 ] -> Bool (float_of_int i1 > i2)
   | G.Gt, [ Float i1; Int i2 ] -> Bool (i1 > float_of_int i2)
+  | G.Gt, [ Date (y1, m1, d1); Date (y2, m2, d2) ] ->
+      Bool (date_compare (y1, m1, d1) (y2, m2, d2) =*= 1)
```

Comment:
We could avoid explicitly deconstructing the args here, since we just pass it into `date_compare`. 

Extra:
If you bound it to say, `d1`, and `d2`, we could also get freaky fancy and write:
```
eval_op G.Gt [ Int (date_compare d1 d2); Int 0 ]
```
and so on, in the other cases.

I'll leave the `eval_op` thing to your judgement, though. I could go either way on that.

Reply:
Switched to using float representation as @aryx suggested, so no longer have separate date representation